crates/holochain/src/core/workflow/countersigning_workflow/accept.rs

```diff
@@ -80,7 +80,7 @@ pub async fn accept_countersigning_request(
     // session request in the workspace.
     let put_accepted_result = workspace.unwrap().inner.share_mut(|inner, _| {
         if inner.session.is_some() {
-            return Err(KitsuneError::other("Session already exists"));
+            return Err(ShareError::ClosureFailed("Session already exists".into()));
```

Comment:
This type is just ignored in the `is_err()` below.

```suggestion
            return Err(());
```

Reply:
But that closure takes a `ShareResult` (see [here](https://github.com/holochain/holochain/blob/44cf01dc3271cc3370de6783ab887b62fee2ef67/crates/holochain/src/core/share.rs#L77)) so your suggested change leads to a compiler error: `expected ShareError, found ()`

Reply:
Oh, that's a very weird api. The closure shouldn't return a share result, but just R, and then the implementor of the closure could choose to return a sub-Result or not.

Okay, well, I'll change my result to an approve, and you can decide if you want to just keep it this way for expediency, or alter all the usages of it : )

Reply:
It is a weird API indeed. But without a Result in the closure, how would you deal with the [closed case](https://github.com/holochain/holochain/blob/44cf01dc3271cc3370de6783ab887b62fee2ef67/crates/holochain/src/core/share.rs#L81)?

Reply:
It would return the closed right away not even invoking the closure.